repositories/db_executor.go

```diff
@@ -36,3 +38,20 @@ func (e ExecutorPostgres) Query(ctx context.Context, sql string, args ...any) (p
 func (e ExecutorPostgres) QueryRow(ctx context.Context, sql string, args ...any) pgx.Row {
 	return e.exec.QueryRow(ctx, sql, args...)
 }
+
+type SqlBuilder interface {
+	ToSql() (string, []interface{}, error)
+}
+
+func ExecBuilder(ctx context.Context, exec Executor, builder SqlBuilder) (rowsAffected int64, err error) {
+	query, args, err := builder.ToSql()
+	if err != nil {
+		return 0, errors.Wrap(err, "can't build sql query")
+	}
+
+	tag, err := exec.Exec(ctx, query, args...)
+	if err != nil {
+		return 0, errors.Wrap(err, fmt.Sprintf("error executing sql query: %s", query))
+	}
+	return tag.RowsAffected(), nil
+}
```

Comment:
I would be in favour of removing this helper :
- It introduce an abstraction that makes code less understandable when reading repositories methods
- It seems the `tag.RowsAffected()` value is always ignored

We can keep it but IMO we need to either :
- moving it as an `Executor` method like this was done before (basically this is a sugar function that aims to directly inject a `SqlBuilder` in place of the `query`)
- move it in a separate file (this is more or less the same kind of stuff as `sql_to_row` `sql_to_models` : a utility function)

Reply:
I'd still rather keep it because it removes tons of boilerplate in all repo methods, but fine with moving it to elsewhere (probably where the SqlToRow etc are).
Not a fan of making it an Executor Method because it makes the interface more complex.

Reply:
TBH, from the time I first wrote this comment to the end of the github review, I finally forged my opininion and I'm aligned with you.